runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java

```diff
@@ -639,6 +660,118 @@ else if (event.getMessage().contains(KILLED_TELEBLOCK_OPPONENT_TEXT))
 				}
 			}
 		}
+
+		if (config.showTzhaarTimers())
+		{
+			String message = Text.removeTags(event.getMessage());
+			Matcher matcher = TZHAAR_COMPLETE_MESSAGE.matcher(message);
+
+			if (message.contains(TZHAAR_DEFEATED_MESSAGE) || matcher.matches())
+			{
+				removeTzhaarTimer();
+				resetTzhaarConfig();
+				resetTzhaarVars();
+				return;
+			}
+
+			Instant now = Instant.now();
+			matcher = TZHAAR_PAUSED_MESSAGE.matcher(message);
+			if (matcher.matches())
+			{
+				tzhaarLastTime = now;
+				createTzhaarTimer(tzhaarStartTime, now);
+				return;
+			}
+
+			matcher = TZHAAR_WAVE_MESSAGE.matcher(message);
+			if (!matcher.matches())
+			{
+				return;
+			}
+
+			if (tzhaarStartTime == null)
+			{
+				int wave = Integer.parseInt(matcher.group(1));
+				if (wave != 1)
+				{
+					return;
+				}
+
+				tzhaarStartTime = now;
+				createTzhaarTimer(tzhaarStartTime, tzhaarLastTime);
+			}
+			else if (tzhaarLastTime != null)
+			{
+				tzhaarStartTime = tzhaarStartTime.plus(Duration.between(tzhaarStartTime, now)).minus(Duration.between(tzhaarStartTime, tzhaarLastTime));
+				tzhaarLastTime = null;
+				createTzhaarTimer(tzhaarStartTime, tzhaarLastTime);
+			}
+		}
+	}
+
+	private void updateTzhaarTimerOnLoading()
+	{
+		if (tzhaarTimer == null)
+		{
+			return;
+		}
+
+		if (!checkInFightCaves() && !checkInInferno())
+		{
+			removeTzhaarTimer();
+			resetTzhaarConfig();
+			resetTzhaarVars();
+		}
+	}
+
+	@VisibleForTesting
+	boolean checkInFightCaves()
+	{
+		return client.getMapRegions() != null && ArrayUtils.contains(client.getMapRegions(), FIGHT_CAVES_REGION_ID);
+	}
+
+	private boolean checkInInferno()
+	{
+		return client.getMapRegions() != null && ArrayUtils.contains(client.getMapRegions(), INFERNO_REGION_ID);
+	}
+
+	private void resetTzhaarVars()
+	{
+		tzhaarStartTime = null;
+		tzhaarLastTime = null;
+	}
+
+	private void removeTzhaarTimer()
+	{
+		infoBoxManager.removeInfoBox(tzhaarTimer);
+		tzhaarTimer = null;
+	}
+
+	private void createTzhaarTimer(Instant startTime, Instant lastTime)
```

Comment:
This is always called with `tzhaarStartTime` and `tzhaarLastTime` as its arguments. Hence, these can be referenced directly instead of being passed as arguments.